### PR TITLE
Expose query/topic builder to `spine-web` API.

### PR DESCRIPTION
**Pull request by YegorUdovchenko**
### Summary
In this PR the API of `spine-web` NPM library was expanded with methods for building and execution of query and subscription topic. The existing methods were composed into `fetch`, `subscribe` and `sendCommand`. Also, the public API was spared from the usage of type and message wrappers (`Type`, `TypedMessage`).

#### `newQuery()`
Creates a new `QueryFactory` for creating `Query` instances specifying the data to be retrieved from Spine server. To build query specifying entities of a developer-defined `Task` type by IDs:
```
newQuery().select(Task)
          .byIds([taskId1, taskId2])
          .build()
```
To execute the resulting `Query` instance pass it to the`Client#execute()` method.

#### `execute(query)`
Executes the given `Query` instance specifying the data to be retrieved from Spine server fulfilling a returned promise with an array of received objects.

#### `newTopic()`
Creates a new `TopicFactory` for building subscription topics specifying the state changes to be observed from Spine server. To build a subscription topic specifying entities of a developer-defined `UserTasks`
type where tasks count is greater than 3:
```
newTopic().select(UserTasks)
          .where(Filters.gt('tasksCount', 3))
          .build()
```
#### `subscribeTo(topic)`
Creates a subscription to the topic which is updated with backend changes.  Fulfills a returning promise with an object representing a result of the subscription to entities state changes.

### CODE REVIEW

**Comment by YegorUdovchenko** [comment]
@armiol PTAL

File: client-js/main/client/abstract-client.js [hunk 1]
```diff
@@ -120,47 +98,34 @@ export class AbstractClient extends Client {
   /**
    * @inheritDoc
    */
-  subscribeToEntities({ofType: type, byIds: ids, byId: id}) {
-    if (typeof ids !== 'undefined' && typeof id !== 'undefined') {
-      throw new Error('No entity IDs set. Specify either a single entity ID or' +
-          ' multiple entity IDs to subscribe to the entity state updates.');
+  fetch({entity: cls, byIds: ids, byId: id}) {
```
Comment (armiol) [inline client-js/main/client/abstract-client.js]: Having both `byId` and `byIds` looks confusing to me. Can we leave just a `byIds` clause and detect whether an array or a single value is passed?

File: client-js/main/client/abstract-client.js [hunk 2]
```diff
@@ -120,47 +98,34 @@ export class AbstractClient extends Client {
...
-   *
-   * @param {!spine.client.Topic} topic a topic of a subscription
-   * @return {Promise<EntitySubscriptionObject>}
-   * @protected
-   * @abstract
+   * @inheritDoc
    */
-  _subscribeTo(topic) {
-    throw new Error('Not implemented in abstract base.');
+  subscribe({entity: cls, byIds: ids, byId: id}) {
```
Comment (armiol) [inline client-js/main/client/abstract-client.js]: Same as above.

File: client-js/main/client/actor-request-factory.js [hunk 3]
```diff
@@ -42,11 +42,21 @@ import {Type, TypedMessage, isProtobufMessage} from './typed-message';
...
 
+// TODO:2019-06-07:yegor.udovchenko: Cover `Filters` class withunit tests
+// https://github.com/SpineEventEngine/web/issues/100
 /**
  * A factory for `Filter` and `CompositeFilter` instances.
  */
 export class Filters {
 
+  /**
+   * @typedef {string | number | boolean | TypedMessage<T> | <T extends Message>} FieldValue
```
Comment (armiol) [inline client-js/main/client/actor-request-factory.js]: We also have time-based values. What if a JS `Date`-typed value is passed? We need to convert it into a `Timestamp` message.

File: client-js/main/client/client.js [hunk 4]
```diff
@@ -31,133 +36,141 @@
...
+   * the state changes to be observed from Spine server.
    *
    * @example
-   * // Fetch all entities of a developer-defined Task type one-by-one using an observable.
-   * fetchAll({ofType: taskType}).oneByOne().subscribe({
-   *   next(task) { ... },
-   *   error(error) { ... },
-   *   complete() { ... }
-   * })
+   * // Build a subscription topic specifying all entities of a developer-defined `UserTasks` type
```
Comment (armiol) [inline client-js/main/client/client.js]: This comment seems irrelevant to the code snippet below.

File: client-js/main/client/client.js [hunk 5]
```diff
@@ -31,133 +36,141 @@
...
+   * newQuery().select(Task)
+   *           .build()
+   *
    * @example
-   * // Fetch all entities of a developer-defined Task type at once using a Promise.
-   * fetchAll({ofType: taskType}).atOnce().then(tasks => { ... })
+   * // Build a subscription topic specifying entities of a developer-defined `UserTasks`
+   * // type where tasks count is greater than 3
```
Comment (armiol) [inline client-js/main/client/client.js]: A period at the end of the sentence is missing.

File: client-js/main/client/client.js [hunk 6]
```diff
@@ -31,133 +36,141 @@
...
-   *   complete() { ... }
-   * })
+   * // Build a subscription topic specifying all entities of a developer-defined `UserTasks` type
+   * newQuery().select(Task)
+   *           .build()
+   *
    * @example
-   * // Fetch all entities of a developer-defined Task type at once using a Promise.
-   * fetchAll({ofType: taskType}).atOnce().then(tasks => { ... })
+   * // Build a subscription topic specifying entities of a developer-defined `UserTasks`
```
Comment (armiol) [inline client-js/main/client/client.js]: Also, let's not use the `developer` term. It's a type of the domain entity. So I would say 
```
 Build a subscription topic for `UserTasks` domain entity, selecting the instances which task count is greater than 3.
```

File: client-js/main/client/client.js [hunk 7]
```diff
@@ -31,133 +36,141 @@
...
+   * // type where tasks count is greater than 3
+   * newTopic().select(UserTasks)
+   *           .where(Filters.gt('tasksCount', 3))
+   *           .build()
    *
-   * @param {!Type<T>} ofType a type of the entities to be queried
-   * @return {Client.Fetch<T>} a fetch object allowing to specify additional remote
-   *                                call parameters and executed the query.
+   * To execute the resulting `Topic` instance pass it to the {@link Client#subscribeTo()}
```
Comment (armiol) [inline client-js/main/client/client.js]: To turn the resulting `Topic` instance into a subscription pass it to the {@link Client#subscribeTo()}.

File: client-js/main/client/client.js [hunk 8]
```diff
@@ -31,133 +36,141 @@
...
    */
-  fetchAll({ofType: type}) {
+  newTopic() {
     throw new Error('Not implemented in abstract base.');
   }
 
   /**
-   * Fetches a single entity of the given type.
+   * Creates a subscription to the topic which is updated with backend changes.
+   * Fulfills a returning promise with an object representing a result of the
```
Comment (armiol) [inline client-js/main/client/client.js]: Let's not re-iterate what a subscription is over and over again. It should be shorter:

```
 Fulfills a returning promise with the created subscription.
```

Same below.

File: client-js/main/client/client.js [hunk 9]
```diff
@@ -203,34 +216,81 @@ export class Client {
   }
 
   /**
-   * Subscribes to entity changes on the backend, providing the changes via `itemAdded`,
-   * `itemChanged`, and `itemRemoved` observers.
+   * Fetches entities of the given class type from the Spine backend. Fetches
```
Comment (armiol) [inline client-js/main/client/client.js]: Let's shorten:
```
Fetches entities of the given class type from the Spine backend. 

Optionally accepts entity identifiers targeting the objects to fetch. If no identifiers are passed, returns all entities of the selected type.
 
The returned promise is fulfilled an array of objects, each representing an entity.

This API call is a shortcut for {@link Client#newQuery()} followed by  {@link Client#execute()}. It covers the most common use cases. If a more advanced fetch behaviour is required, the `Query` instance should be created and parameterized via {@link Client#newQuery()}.
```

Please rewrite the similar pieces in this file in the same way.

**Comment by YegorUdovchenko** [comment]
@armiol PTAL

File: client-js/main/client/abstract-client.js [hunk 10]
```diff
@@ -120,47 +98,46 @@ export class AbstractClient extends Client {
...
+   * Builds target from the given target builder specifying the set
+   * of target entities.
    *
-   * @param {!spine.client.Topic} topic a topic of a subscription
-   * @return {Promise<EntitySubscriptionObject>}
-   * @protected
-   * @abstract
+   * The built target points to:
+   *  - the particular entities with the given IDs;
+   *  - the all entities if no IDs specified;
```
Comment (armiol) [inline client-js/main/client/abstract-client.js]: `all the entities of the specified type, if no IDs specified`

File: client-js/main/client/abstract-client.js [hunk 11]
```diff
@@ -120,47 +98,46 @@ export class AbstractClient extends Client {
...
   /**
-   * Creates a subscription to the topic which is updated with backend changes.
+   * Builds target from the given target builder specifying the set
+   * of target entities.
    *
-   * @param {!spine.client.Topic} topic a topic of a subscription
-   * @return {Promise<EntitySubscriptionObject>}
-   * @protected
-   * @abstract
+   * The built target points to:
```
Comment (armiol) [inline client-js/main/client/abstract-client.js]: `The resulting target points to`

File: client-js/main/client/abstract-client.js [hunk 12]
```diff
@@ -120,47 +98,46 @@ export class AbstractClient extends Client {
...
+   *  - the particular entities with the given IDs;
+   *  - the all entities if no IDs specified;
+   *
+   * @param {AbstractTargetBuilder<Query|Topic>} targetBuilder
+   *    a builder for creating `Query` or `Topic` instances.
+   * @param {?<T extends Message>[] | <T extends Message> | Number[] | Number | String[] | String} ids
+   *      a list of target entities IDs or an ID of a single target entity
+   * @return {Query|Topic} the built target
+   *
+   * @template <T> a class of a query or subscription target entities identifiers
```
Comment (armiol) [inline client-js/main/client/abstract-client.js]: `a class of identifiers, corresponding to the query/subscription targets`

File: client-js/main/client/actor-request-factory.js [hunk 13]
```diff
@@ -42,11 +42,21 @@ import {Type, TypedMessage, isProtobufMessage} from './typed-message';
 import {AnyPacker} from './any-packer';
 import {FieldPaths} from './field-paths';
 
+// TODO:2019-06-07:yegor.udovchenko: Cover `Filters` class withunit tests
```
Comment (armiol) [inline client-js/main/client/actor-request-factory.js]: `...with the unit...`

**Comment by armiol** [comment]
@YegorUdovchenko also, please fix the build before merging.
